services/frontend-service/src/legacy-ui/App/AuthContext.tsx

```diff
@@ -0,0 +1,126 @@
+import { ConfigsContext } from './index';
+import { Configuration, PublicClientApplication } from '@azure/msal-browser';
+import {
+    MsalProvider,
+    AuthenticatedTemplate,
+    useIsAuthenticated,
+    UnauthenticatedTemplate,
+    useMsal,
+} from '@azure/msal-react';
+
+type AuthContextType = {
+    useAzureAuth: boolean;
+    useAuth: boolean;
+};
+
+const AuthContext = React.createContext<AuthContextType>({} as AuthContextType);
+
+const getMsalConfig = (configs: any): Configuration => ({
+    auth: {
+        clientId: configs?.authConfig?.azureAuth?.clientId,
+        authority: `${configs?.authConfig?.azureAuth?.cloudInstance}${configs?.authConfig?.azureAuth?.tenantId}`,
+        redirectUri: configs?.authConfig?.azureAuth?.redirectURL,
+    },
+    cache: {
+        cacheLocation: 'sessionStorage',
+        storeAuthStateInCookie: false,
+    },
+});
+
+const getLoginRequest = () => ({
+    scopes: ['User.Read'],
+});
+
+type AuthTokenContextType = {
+    token: String;
+    authHeader: {
+        Authorization?: String;
+    };
+};
+
+export const AuthTokenContext = React.createContext<AuthTokenContextType>({} as AuthTokenContextType);
+
+function AzureAuthTokenProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const [token, setToken] = React.useState('');
+    const [authHeader, setAuthHeader] = React.useState({});
+    const { instance, accounts } = useMsal();
+
+    React.useEffect(() => {
+        const request = {
+            ...loginRequest,
+            account: accounts[0],
+        };
+        instance
+            .acquireTokenSilent(request)
+            .then((response) => {
+                setToken(response.accessToken);
+                setAuthHeader({ Authorization: response.accessToken });
+            })
+            .catch(() => {
+                instance.acquireTokenPopup(request).then((response) => {
+                    setToken(response.accessToken);
+                    setAuthHeader({ Authorization: response.accessToken });
+                });
+            });
+    }, [instance, accounts, loginRequest]);
+    return <AuthTokenContext.Provider value={{ token, authHeader }}>{children}</AuthTokenContext.Provider>;
+}
+
+const AzureAutoSignIn = () => {
+    const isAuthenticated = useIsAuthenticated();
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const { instance } = useMsal();
+    if (!isAuthenticated) {
+        instance.loginRedirect(loginRequest);
+    }
+    return <></>;
+};
+
+export function AuthProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const { configs } = React.useContext(ConfigsContext);
+    const useAzureAuth = configs?.authConfig?.azureAuth?.enabled || false;
+    const useAuth = useAzureAuth;
+    const msalConfig = React.useMemo(() => getMsalConfig(configs), [configs]);
+    const msalInstance = new PublicClientApplication(msalConfig);
```

Comment:
Is It ok to send ` auth: {
        clientId: null,
        authority:  null,
        redirectUri: null,
    },
    cache: {
        cacheLocation: 'sessionStorage',
        storeAuthStateInCookie: false,
    },` to `new PublicClientApplication()` 
 Please add some tests for `AuthProvider`